### ISSUE TITLE
Metadata is not read when playing SHOUTcast audio stream

### ISSUE THREAD

**Issue by evgenii-m** [caprica/vlcj#543]
Hi!
I use HeadlessMediaPlayer for playing audio from SHOUTcast stream and it works, but i can't read the metadata from this stream. The code I use:

```
package ru.push.caudioplayer;

import uk.co.caprica.vlcj.player.MediaMeta;
import uk.co.caprica.vlcj.player.MediaPlayer;
import uk.co.caprica.vlcj.player.MediaPlayerFactory;

import java.io.IOException;

public class AudioMetadataTest {

  public static void main(String[] args) throws IOException {
    String[] mediaPlayerOptions = {};
    MediaPlayerFactory mediaPlayerFactory = new MediaPlayerFactory(mediaPlayerOptions);
    MediaPlayer mediaPlayer = mediaPlayerFactory.newHeadlessMediaPlayer();

    String resourceUri = "http://160.75.86.29:8090/";
    boolean playMediaResult = mediaPlayer.playMedia(resourceUri);
    mediaPlayer.parseMedia();

    if (playMediaResult) {
      System.out.println("Press any key for read metadata.");
      System.in.read();
      MediaMeta mediaMeta = mediaPlayer.getMediaMeta();
      System.out.println("Artist: " + mediaMeta.getArtist());
      System.out.println("Title: " + mediaMeta.getTitle());
      System.out.println("Album: " + mediaMeta.getAlbum());
      System.out.println("Genre: " + mediaMeta.getGenre());
      System.out.println("URL: " + mediaMeta.getUrl());
      System.out.println("Now playing: " + mediaMeta.getNowPlaying());
    } else {
      System.out.println("Play audio fails");
    }

    System.out.println("Press any key for stop audio player.");
    System.in.read();
    mediaPlayer.stop();
    mediaPlayer.release();
    mediaPlayerFactory.release();
  }
}
```

Program output:
```
Press any key for read metadata.
[0x7f6a840041e8] access_http access: Raw-audio server found, mp3 demuxer selected
[0x7f6a740147d8] mpgatofixed32 audio converter error: libmad error: bad main_data_begin pointer
[0x7f6a740147d8] mpgatofixed32 audio converter error: libmad error: bad main_data_begin pointer

Artist: null
Title: ITU Radio Jazz/Blues
Album: null
Genre: Jazz/Blues
URL: null
Now playing: null
Press any key for stop audio player.
```

I use VLCJ v.3.10.1. 
In the VLC player (and other audio players) metadata from this stream is got correctly, the current artist and the song name are ok.

**Comment by caprica** [comment]
The synchronous parseMedia() call can sometimes cause hangs when used on streams.

Having said that, I don't know in your case. Try waiting for media meta changed events on the listener and see if the meta data is coming 'late' - i.e. the meta data might not be available until some of the stream has already been consumed.

**Comment by evgenii-m** [comment]
I added listener for the mediaMetaChanged event, but this didn't help. Code fragment:
```
    String resourceUri = "http://160.75.86.29:8090/";
    mediaPlayer.playMedia(resourceUri);
    mediaPlayer.parseMedia();
    mediaPlayer.addMediaPlayerEventListener(new MediaPlayerEventAdapter() {
      @Override
      public void mediaMetaChanged(MediaPlayer mediaPlayer, int metaType) {
        MediaMeta mediaMeta = mediaPlayer.getMediaMeta();
        System.out.println("Artist: " + mediaMeta.getArtist());
        System.out.println("Title: " + mediaMeta.getTitle());
        System.out.println("Album: " + mediaMeta.getAlbum());
        System.out.println("Genre: " + mediaMeta.getGenre());
        System.out.println("URL: " + mediaMeta.getUrl());
        System.out.println("Now playing: " + mediaMeta.getNowPlaying());
      }
    });
```
Program output:
```
Artist: null
Title: ITU Radio Jazz/Blues
Album: null
Genre: Jazz/Blues
URL: null
Now playing: null
```
Any ideas? Maybe special options are needed for mediaPlayerFactory?

**Comment by evgenii-m** [comment]
I realized the problem - it is necessary to access the audio stream at a different URL. If I connect to this stream at different URL (http://160.75.86.29:8088/stream/3), I will get the information about the current track by mediaMeta.getNowPlaying().
Thank you for the attention.

**Comment by evgenii-m** [comment]
It's interesting that in VLC player and other audio players reading meta data from this stream are also available at the http://160.75.86.29:8090.

**Comment by caprica** [comment]
Probably your original MRL is a playlist, and the actual meta data is attached to the first sub-item in the playlist.
